### PR TITLE
Fix null entityId in audit log when alarm is deleted

### PR DESCRIPTION
**Pull request by irynamatveieva**
## Pull Request description

- changed types Deleted -> Alarm deleted, Acknowledged -> Alarm acknowledged, etc.
- added alarmId and originatorId to the audit log details when Type is Alarm deleted.
- changed Entity name to name of entity instead of alarm name when Type is one of alarm action types in the audit logs

PE PR: https://github.com/thingsboard/thingsboard-pe/pull/2767

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

**Review by ShvaykaD — CHANGES_REQUESTED** [review]
Please check my comment above.

File: dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @Test
+    public void givenEntityIsNull_whenLogEntityAction_thenShouldFetchEntityName() {
+        // GIVEN
+        given(auditLogLevelFilter.logEnabled(any(), any())).willReturn(true);
+
+        // WHEN
+        auditLogService.logEntityAction(TENANT_ID, CUSTOMER_ID, USER_ID, USER_NAME, DEVICE_ID, null, ActionType.ADDED, null);
+
+        // THEN
+        then(entityService).should().fetchEntityName(TENANT_ID, DEVICE_ID);
```
Comment (ShvaykaD) [inline dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java:84]: please verify entity name in auditLogEntry.

File: dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    private final String USER_NAME = "Test User";
+
+    @InjectMocks
+    private AuditLogServiceImpl auditLogService;
+    @Mock
+    private EntityService entityService;
+    @Mock
+    private AuditLogLevelFilter auditLogLevelFilter;
+    @Mock
+    private AuditLogDataValidator auditLogDataValidator;
```
Comment (ShvaykaD) [inline dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java:67]: unused. Please make it used.

File: dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    @Mock
+    private JpaExecutorService executor;
+    @Mock
+    private AuditLogDao auditLogDao;
+    @Mock
+    private AuditLogSink auditLogSink;
```
Comment (ShvaykaD) [inline dao/src/test/java/org/thingsboard/server/dao/audit/AuditLogServiceImplTest.java:73]: unused. Please make it used.

**Review by ShvaykaD — CHANGES_REQUESTED** [review]

File: common/data/src/main/java/org/thingsboard/server/common/data/audit/ActionType.java [hunk 4]
```diff
@@ -22,52 +22,72 @@
...
+    UPDATED(TbMsgType.ENTITY_UPDATED), // log entity
+    ATTRIBUTES_UPDATED(TbMsgType.ATTRIBUTES_UPDATED), // log attributes/values
+    ATTRIBUTES_DELETED(TbMsgType.ATTRIBUTES_DELETED), // log attributes
+    TIMESERIES_UPDATED(TbMsgType.TIMESERIES_UPDATED), // log timeseries update
+    TIMESERIES_DELETED(TbMsgType.TIMESERIES_DELETED), // log timeseries
+    RPC_CALL(), // log method and params
+    CREDENTIALS_UPDATED(), // log new credentials
+    ASSIGNED_TO_CUSTOMER(TbMsgType.ENTITY_ASSIGNED), // log customer name
+    UNASSIGNED_FROM_CUSTOMER(TbMsgType.ENTITY_UNASSIGNED), // log customer name
+    ACTIVATED(), // log string id
```
Comment (ShvaykaD) [inline common/data/src/main/java/org/thingsboard/server/common/data/audit/ActionType.java:36]:
```suggestion
    ACTIVATED, // log string id
```

**Review by ShvaykaD — CHANGES_REQUESTED** [review]

**Review by ShvaykaD — CHANGES_REQUESTED** [review]
Hi @irynamatveieva, there are some tests that fail in the Alarm Controller Test. Looks related to the changes added in the PR.

**Review by ViacheslavKlimov — APPROVED** [review]

**Pull request merged**